django_modern_rest/internal/middleware_wrapper.py

```diff
@@ -0,0 +1,118 @@
+import inspect
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, TypeAlias, TypeVar
+
+from django.http import HttpRequest, HttpResponse
+
+if TYPE_CHECKING:
+    from django_modern_rest.response import ResponseDescription
+
+TypeT = TypeVar('TypeT', bound=type[Any])
+_CallableAny: TypeAlias = Callable[..., Any]
+MiddlewareDecorator: TypeAlias = Callable[[_CallableAny], _CallableAny]
+ResponseConverter: TypeAlias = Callable[[HttpResponse], HttpResponse]
+_ConverterSpec: TypeAlias = tuple['ResponseDescription', ResponseConverter]
+_ViewDecorator: TypeAlias = Callable[[_CallableAny], _CallableAny]
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class DecoratorWithResponses:
+    """Type for decorator with responses attribute."""
+
+    decorator: Callable[[Any], Any]
```

Comment:
```suggestion
    decorator: Callable[[_TypeT], _TypeT]
```

Let's make this class generic.